SConstruct

```diff
@@ -141,6 +142,16 @@ def RecursiveGlob(pattern, dir_name=buildDirectory):
 	matches = [i for i in matches if not '{}main.cpp'.format(os.path.sep) in str(i)]
 	return matches
 
+try:
+    env.Tool('compilation_db')
+    env.CompilationDatabase()
+    env.Default('compile_commands.json')
```

Comment:
We don't need to generate this in CI builds

Reply:
Do you mean we should not add it to the default target? It's pretty convenient to have it generate when building and it takes like less than a second to create so not that big of a deal imo.

Reply:
FWIW any build that selects a target will not regenerate this file (because SCons is not building the default target then).
So `scons endless-sky` or `scons test` will not (re)build it, only `scons` (and things like `scons mode=debug`).

We can also avoid depending on the default name, or needing to restate the name (should we choose to implement mode-based specific files):
```python
    env.Default(env.CompilationDatabase())
```